src/repositories/CourseRepository.js

```diff
@@ -24,11 +24,12 @@ exports.findCourseAndPopulate = (query, origin, populateVirtual = false) => Cour
     ...(origin === WEBAPP
       ? [
         { path: 'trainer', select: 'identity.firstname identity.lastname' },
-        {
+        ...(query.accessRules ? [{
```

Comment:
Est-ce necessaire de retourner un tableau vide si il n'y a pas d'accessRules dans la query ? On ne pourrait pas fair un : 
```
...(query.accessRules && [{
          path: 'trainees',
          select: '_id company',
          populate: { path: 'company', populate: { path: 'company', select: 'name' } },
        }]
```

Reply:
Je n'ai pas l'impression que ce soit possible avec les tableaux : https://dev.to/boywithsilverwings/conditionally-spread-into-object-array-javascript-3h5i

Reply:
Ah yes